C/jets.c

```diff
@@ -589,6 +589,72 @@ RIGHT_EXTEND_(16,32)
 RIGHT_EXTEND_(16,64)
 RIGHT_EXTEND_(32,64)
 
+#define LEFT_SHIFT_(log, bits)                                                           \
+static inline void left_shift_helper_##bits(bool with, frameItem* dst, frameItem *src) { \
+  uint_fast8_t amt = read##log(src);                                                     \
```

Comment:
1f245eccc7593859bcdcbfe24d0c09f4c8443867 There will be a compiler error if `log > 8`. This will not break any binaries, but it might be confusing.

Reply:
I fail to see why there is a compiler error.

But yeah, this code assumes that `log` <= 64 (because otherwise truncation will happen.) Perhaps that should be asserted.